apps/tx-builder/src/components/forms/SolidityForm.tsx

```diff
@@ -197,11 +263,12 @@ const SolidityForm = ({
         )}
 
         {/* Switch button to encoding contract fields values to hex data */}
-        <Text size="lg">
-          <Switch checked={showHexEncodedData} onChange={onClickShowHexEncodedData} />
-          Use custom data (hex encoded)
-        </Text>
-
+        {!hideHexToggler && (
```

Comment:
This prop name is a little confusing IMO as is hide instead show and is used with a NOT (!). As well is declared as nullable. Perhaps could be better to call it `showHexToggler` instead negating it ?